source/testcommon/README.md

```diff
@@ -1,9 +1,149 @@
-# Test Common Py
+# Testcommon utility tool
 
-This is common testing utilities for Python projects.
+This python packages includes tools and utility function for making testing easier.
+
+## Usage
+
+### Scenario testing
+
+To use testcommon for the scenario testing we need to set up three things.
+
+---
+
+#### 1) Use the standard folder structure setup
+
+```plaintext
+├── scenario_name/
+    ├── when/
+        ├── input1.csv
+        └── ...
+    └── then/
+        ├── output1.csv
+        └── ...
+    └── test_output.py
+└── conftest.py
+```
+
+Input files are stored in the `/when` subfolder, while the expected results are stored in the `/then` subfolder.
+We prefix scenario names with `given_...` to enhance readability. As if we ready it line-by-line it makes sense.
+This is best shown with an example.
+
+```plaintext
+├──given_energy_scenario
+    ├── when/
+        ├── energy.csv
+        ├── energy_per_brp.csv
+        └── energy_per_es.csv
+    └── then/
+        └── energy_v1.csv
+```
+
+Notice, that there is not a one-to-one relation between the `/then` and `/when` files.
+This is because we may have to perform various transformations, such as aggregating or joining tables, to achieve the
+desired
+output. (i.e., `/then` file).
+---
+
+#### 2) conftest.py
+
+The test setup might seem unusual at first glance, as the individual test files do not perform any transformations
+but solely compare actual results to expected results. The idea behind using `fixtures` is that many of the scenario
+tests requires an identical setup (i.e. catalogs, schemas, and tables). Building and tearing down these setups for every
+test would waste a lot of time and resources. Additionally, specifying everything within each test would lead to code
+duplication and make the tests hard to read and understand. By substituting the logic and transformations into
+the `conftest.py` (a special pytest file where `fixtures` can be defined), we can achieve code simplicity, code
+reusability, and reduced test execution time, especially for long-running transformations. `Fixtures` are
+functions that run before the actual tests, allowing us to set up the environment, perform calculations, and execute
+logic and transformations before any tests are run. We can control when the `fixtures` are set up
+and torn down by setting the `scope` parameter.
+
+These are different `scope` types:
+
+1. Function: the default scope, the fixture is destroyed at the end of the test.
+1. Class: the fixture is destroyed during teardown of the last test in the class.
+1. Module: the fixture is destroyed during teardown of the last test in the module.
+1. Package: the fixture is destroyed during teardown of the last test in the package.
+1. Session: the fixture is destroyed at the end of the test session.
+
+In the example below, we are using the `module` scope. This means that all the tests within a module will reuse the same
```

Comment:
"In the example below". We'll (almost?) everytime use `module` as all then-files in a scenario depends on the exact same transformation. And scenarios should always be independent of each other. This leaves only "module" as the single reasonable choice.